freeipa/src/main/java/com/sequenceiq/freeipa/repository/StackRepository.java

```diff
@@ -67,6 +67,11 @@
             + "LEFT JOIN FETCH ig.instanceMetaData WHERE s.environmentCrn = :environmentCrn AND s.accountId = :accountId AND s.terminated = -1")
     Optional<Stack> findByEnvironmentCrnAndAccountIdWithList(@Param("environmentCrn") String environmentCrn, @Param("accountId") String accountId);
 
+    @CheckPermission(action = ResourceAction.READ)
+    @Query("SELECT s FROM Stack s LEFT JOIN FETCH s.childEnvironments c LEFT JOIN FETCH s.instanceGroups ig "
+            + "LEFT JOIN FETCH ig.instanceMetaData WHERE c.environmentCrn = :environmentCrn AND s.accountId = :accountId AND s.terminated = -1")
```

Comment:
can we check for the status? I am afraid if somebody change the terminated = -1 we will be broken

Reply:
we work only with non-terminated stacks, which is indicated by this everywhere. I won't join another table for this reason, if not really necessary